src/MatrixClientPeg.js

```diff
@@ -40,6 +40,14 @@ function deviceId() {
 class MatrixClientPeg {
     constructor() {
         this.matrixClient = null;
+
+        // These are the default options used when Lifecycle.js
+        // starts the client. These can be altered when the
+        // 'will_start_client' event is dispatched.
+        this.opts = {
```

Comment:
is there a reason this is here rather than in `Lifecycle`?

the globalness and mutableness of it make me unhappy